ibm/resource_ibm_container_alb_create.go

```diff
@@ -0,0 +1,198 @@
+			},
+			"alb_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				//ForceNew:    true,
+				Description: "The type of ALB that you want to create.",
+			},
+			"vlan_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				//ForceNew:    true,
+				Description: "The VLAN ID that you want to use for your ALBs.",
+			},
+			"zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The zone where you want to deploy the ALB.",
+			},
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the cluster that the ALB belongs to.",
+			},
+
+			//response
+			"alb_id": {
+				Type: schema.TypeString,
+				//Required:    false,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The ID of the application load balancer (ALB).",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ALB name",
+			},
+			"disable_deployment": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"enable"},
+				Description:   "Set to true if ALB needs to be disabled",
+			},
+			"user_ip": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "IP assigned by the user",
+			},
+			"replicas": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "number of instances",
+			},
+			"resize": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "resize",
+			},
+		},
+	}
+}
+
+func resourceIBMContainerClassicAlbCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Println("resourceIBMContainerClassicAlbCreate")
+	albClient, err := meta.(ClientSession).ContainerAPI()
+	if err != nil {
+		return err
+	}
+	albAPI := albClient.Albs()
+
+	// "cluster":"string" //mandatory
+	// "enableByDefault": true,
+	// "ingressImage": "string",
+	// "ip": "string",
+	// "nlbVersion": "string",
+	// "type": "string", //mandatory
+	// "vlanID": "string", //mandatory
+	// "zone": "string" //mandatory
+
+	var cluster string
+	if v, ok := d.GetOkExists("cluster"); ok {
+		cluster = v.(string)
+	} else {
+		return fmt.Errorf("Provide `clusterIDorName`")
+	}
+
+	var albType string
+	if v, ok := d.GetOkExists("alb_type"); ok {
+		albType = v.(string)
+	} else {
+		return fmt.Errorf("Provide `alb_type`")
+	}
+
+	var vlanID string
+	if v, ok := d.GetOkExists("vlan_id"); ok {
+		vlanID = v.(string)
+	} else {
+		return fmt.Errorf("Provide `vlanID`")
+	}
+
+	var zone string
+	if v, ok := d.GetOkExists("zone"); ok {
+		zone = v.(string)
+	} else {
+		return fmt.Errorf("Provide `zone`")
+	}
+
+	enableByDefault := d.Get("enable").(bool)
```

Comment:
For optional fields you need to use d.GetOk()
For required variables you can directly use d.Get()